src/Microsoft.DotNet.Build.Tasks/ExecWithRetriesForNuGetPush.cs

```diff
@@ -0,0 +1,125 @@
+        /// because the package "already exists" on the next response.  This indicates that the
+        /// timeout occurred, but the push was actually successful.
+        ///
+        /// To address the condition, I'm implementing a special "IgnoredErrorMessagesWithConditional"
+        /// property that allows you to specify error messages which you want to ignore.  If you
+        /// specify the "ConditionalErrorMessage" metadata on the Item, then the error message is
+        /// only ignored if the "conditional" error message was detected in a previous (or current)
+        /// Exec attempt.
+        /// </summary>
+        public ITaskItem[] IgnoredErrorMessagesWithConditional { get; set; }
+
+        private CancellationTokenSource _cancelTokenSource = new CancellationTokenSource();
+
+        private Exec _runningExec;
+
+        public override bool Execute()
+        {
+            HashSet<string> activeIgnorableErrorMessages = new HashSet<string>();
+            // Add any "Ignore" messages that don't have conditionals to our active list.
+            if (IgnoredErrorMessagesWithConditional != null)
+            {
+                foreach(var message in IgnoredErrorMessagesWithConditional)
+                {
+                    string conditional = message.GetMetadata("ConditionalErrorMessage");
+                    if (string.IsNullOrEmpty(conditional))
+                    {
+                        activeIgnorableErrorMessages.Add(message.ItemSpec);
+                    }
+                }
+            }
+            for (int i = 0; i < MaxAttempts; i++)
+            {
+                string attemptMessage = $"(attempt {i + 1}/{MaxAttempts})";
+                _runningExec = new Exec
+                {
+                    BuildEngine = BuildEngine,
+                    Command = Command,
+                    LogStandardErrorAsError = false,
+                    IgnoreExitCode = true,
+                    ConsoleToMSBuild = true
+                };
+                if (!_runningExec.Execute())
+                {
+                    Log.LogError("Child Exec task failed to execute.");
+                    break;
+                }
+
+                int exitCode = _runningExec.ExitCode;
+                if (exitCode == 0)
+                {
+                    return true;
+                }
+
+                if (_runningExec.ConsoleOutput != null &&
+                    IgnoredErrorMessagesWithConditional != null &&
+                    _runningExec.ConsoleOutput.Length > 0)
+                {
+                    var consoleOutput = _runningExec.ConsoleOutput.Select(c => c.ItemSpec);
+                    // If the console output contains a "conditional" message, add the item to the active list.
+                    var conditionMessages = IgnoredErrorMessagesWithConditional.Where(m => consoleOutput.Any(n => n.Contains(m.GetMetadata("ConditionalErrorMessage"))));
```

Comment:
I usually chop to get lines < 100 chars, especially when Linq is involved, for readability.